ws/v1.md

```diff
@@ -0,0 +1,112 @@
+
+#### Request messages
+
+##### Subscribe
+
+[See payload schema](https://github.com/0xProject/0x-monorepo/blob/development/packages/json-schemas/schemas/relayer_api_orberbook_channel_subscribe_schema.ts#L1)
+
+Minimum request:
+```
+{
+    "type": "subscribe",
+    "channel": "orders",
+    "requestId": "123e4567-e89b-12d3-a456-426655440000"
+}
+```
+This will subscribe to all new orders and order state changes in the orderbook.
+
+Filtered request:
+```
+{
+    "type": "subscribe",
+    "channel": "orders",
+    "requestId": "123e4567-e89b-12d3-a456-426655440000",
+    "payload": {
+        "makerAssetData": "0x01323b5d4c32345ced77393b3530b1eed0f346429d",
+        "takerAssetData": "0x01ef7fff64389b814a946f3e92105513705ca6b990",
+        "networkId": 42,
+    }
+}
+```
+This will subscribe to all new Kovan orders and Kovan order state changes in the orderbook with `makerAssetData` and `takerAssetData` equal to the values specified in the subscribe request.
+
+**Parameters**
+
+General:
+*   `requestId`: a string uuid that will be sent back by the server in response messages so the client can appropriately respond when multiple subscriptions are made
+*   `networkId`: the Ethereum network id to which you'd like to subscribe. Default is 1 (mainnet). (optional)
+   
+Networks and their Ids:
+
+| Network Id| Network Name |
+| ----------| ------------ |
+| 1         | Mainnet      |
+| 42        | Kovan        |
+| 3         | Ropsten      |
+| 4         | Rinkeby      |
+
+Filter parameters (optional): 
+*   `makerAssetType`: subscribes to new orders where the maker asset is of certain asset type (example: `ERC20`, `ERC721`)
+*   `takerAssetType`:  subscribes to new orders where the taker asset is of certain asset type (example: `ERC20`, `ERC721`)
+*   `makerAssetAddress`: subscribes to new orders where the contract address for the maker asset matches the value specified
+*   `takerAssetAddress`: subscribes to new orders where the contract address for the taker asset matches the value specified
+
+Order specific parameters (optional):
+*   `makerAssetData`: subscribes to new orders with the specified makerAssetData
+*   `takerAssetData`: subscribes to new orders with the specified takerAssetData
+*   `traderAssetData`: subscribes to new orders where either makerAssetData or takerAssetData has the value specified
+
+While there is some redundancy in supporting maker/takerAssetType, maker/takerAssetAddress, and maker/takerAssetData, they are actually all needed. For example, you cannot query "all ERC712 orders", or "all Cryptokitties orders", or "all ERC20 orders" with just maker/takerAssetData and need the other query params to do so.
+
+#### Response messages
+
+##### Update
+
+This message is sent whenever the relayer receives or hears about a new order, or when the `remainingFillableAmount` of an order changes because a fill occurred, or because the order was cancelled (in which case the remaining fillable amount would be 0).
```

Comment:
What about changes in whether or not the maker is able to fill the order? If a maker no longer has sufficient tokens to fill an order, OpenRelay removes it from our orderbook.

Additionally, what about expiration? I'm assuming we can rely on consumers to recognize than order is now expired. We don't proactively prune expirations from our database, we just leave it up to a filter at query time, so we're not actively paying attention to when orders expire.

Reply:
I think if a maker no longer has sufficient tokens to fill a full order then the the `remainingFillableAmount` should denote the maximum amount that a  a taker would be able to get.

In the case of expiration, I think it is fair to expect the consumers of the API to be keeping track of this, as it is information that is available in the response.

I will clarify the different edge cases in the spec.

Reply:
Yeah, some relayers might prune orders once they become partially fillable, others will keep partially fillable orders (e.g when a maker has moved half the order amount to a different account). `remainingFillableAmount` let's Relayers use either strategy and allows traders to get a sense for how much is still fillable from each order.